src/runtime/d3d12compute.cpp

```diff
@@ -73,30 +73,30 @@
 #define HALIDE_D3D12_COMMAND_LIST_TYPE D3D12_COMMAND_LIST_TYPE_DIRECT
 #endif
 
-static void d3d12_debug_dump(error &err);
+WEAK void d3d12_debug_dump();
 
 #define d3d12_panic(...)                               \
     do {                                               \
         error err(nullptr);                            \
         err << __VA_ARGS__ << "\n";                    \
         err << "vvvvv D3D12 Begin Debug Dump vvvvv\n"; \
-        d3d12_debug_dump(err);                         \
+        d3d12_debug_dump();                            \
         err << "^^^^^ D3D12  End  Debug Dump ^^^^^\n"; \
         err << "D3D12 HALT !!!\n";                     \
     } while (0)
 
 // v trace and logging utilities for debugging v
 // ! definitely not super thread-safe stuff... !
 // in case there's no 'user_context' available in the scope of a function:
-static void *const user_context = nullptr;
+static constexpr void *const user_context = nullptr;
```

Comment:
changing this to constexpr is less-bad than what was here before, but may I gently suggest that this symbol is a bad idea and should be removed entirely -- it's is just a workaround for needing a `user_context` in places you don't have one. You should really remove this (in a subsequent PR) and either pass the user_context you need down to all the places it is needed, or figure out how to do without. (Or, in some unusual cases, just pass `nullptr` directly... but at least in those cases the passing of `nullptr` will be obvious, rather than obfuscated.)